### PR TITLE
[Bugfix] Users Bulk Action | Password Confirmation Modal

### PR DESCRIPTION
**Pull request by Civolilah**
@beganovich @turbo124 The PR includes fixes for password-protecting bulk actions routes on the Users table. Quick demo:

https://github.com/invoiceninja/ui/assets/51542191/7ec6451f-7c5a-40ca-bae4-74d32f649b33

Let me know your thoughts.

### CODE REVIEW

**Review by beganovich — COMMENTED** [review]
While this generally works, it seems odd to implement and bake this directly in the datatable. The users table is the only one (for now) that implements this functionality and by looking at it we're bloating the datatable quite a lot.

Is there specific reason you implemented this directly? If not, please take implementation out of the table and proceed as with app conventions.

**Comment by Civolilah** [comment]
> While this generally works, it seems odd to implement and bake this directly in the datatable. The users table is the only one (for now) that implements this functionality and by looking at it we're bloating the datatable quite a lot.
> 
> Is there specific reason you implemented this directly? If not, please take implementation out of the table and proceed as with app conventions.

@beganovich The first approach I initially implemented would allow us to pass just one prop to the DataTable component, and the password protection on bulk routes would be applied to any entity. However, you are correct about the second part: the DataTable component would be less complex if we move it out of that component. I have updated the PR with the new approach, which involves moving the logic out of the DataTable component. Let me know your thoughts.

File: src/common/queries/users.ts [hunk 1]
```diff
@@ -46,3 +49,41 @@ export function useBlankUserQuery() {
...
+      },
+      { headers: { 'X-Api-Password': password } }
+    )
+      .then(() => {
+        toast.success(`${action}d_user`);
+
+        ids.forEach((id) => {
+          queryClient.invalidateQueries('/api/v1/users');
+          queryClient.invalidateQueries(route('/api/v1/users/:id', { id }));
+        });
```
[the remark this answers is not in the document]
Comment (beganovich) [reply src/common/queries/users.ts:80]: Let's wait for that PR merge so you can rebase.

**Comment by Civolilah** [comment]
@beganovich The PR has been updated with changes related to merging the PR for query invalidation.

File: src/common/queries/users.ts [hunk 2]
```diff
@@ -45,3 +49,38 @@ export function useBlankUserQuery() {
     { staleTime: Infinity, enabled: isAdmin }
   );
 }
+
+export function useBulk() {
+  const setLastPasswordEntryTime = useSetAtom(lastPasswordEntryTimeAtom);
+
+  const $refetch = useRefetch();
```
Comment (beganovich) [inline src/common/queries/users.ts:56]: You don't have to use a hook, there's global fn `$refetch`.

**Review by beganovich — COMMENTED** [review]
If I understood this well, does this only work for built-in actions? 
We can't protect custom bulk action with a password, right?

**Comment by turbo124** [comment]
@beganovich this is only required on bulk actions on Users at this point.

**Comment by beganovich** [comment]
@turbo124 correct, but the PR supports only built-in (archive, delete, restore). No support for custom actions. Is that ok?

**Comment by turbo124** [comment]
I believe that is the only ones we have. I do not think the remove action has been added on the User Bulk Actions. @Civolilah can confirm?

**Comment by Civolilah** [comment]
> I believe that is the only ones we have. I do not think the remove action has been added on the User Bulk Actions. @Civolilah can confirm?

@turbo124 Exactly, we only have delete, archive, and restore as bulk actions. If we add any other custom actions, it will be easy to support.

**Comment by beganovich** [comment]
Got it, thanks!

**Pull request merged**